src/github_installations.ml

```diff
@@ -42,3 +39,21 @@ let action_as_github_app ~bot_info ~key ~app_id ~owner ~repo action =
   | None ->
       action_with_new_installation_token ~bot_info ~key ~app_id ~owner ~repo
         action
+
+let action_as_github_app ~bot_info ~key ~app_id ~owner ~repo action =
+  (* Executes an action with an installation token if the repository has
+     the GitHub app installed.
+     Generates a new installation token if the existing one has expired. *)
+  match GitHub_app.make_jwt ~key ~app_id with
+  | Ok jwt -> (
+      GitHub_queries.get_installations
+        ~bot_info:{bot_info with github_token= INSTALL_TOKEN jwt}
```

Comment:
If you put `get_installations` inside `GitHub_app` instead of `GitHub_queries`, you wouldn't have to export the `make_jwt` function and abuse the `INSTALL_TOKEN` constructor to store something that is not an install token.

Furthermore, it seems to me that there is a problem with the order of the calls. Why do a query to GitHub before even checking if we have a token in store for this owner? This is going to add many unnecessary requests.